packages/runtime/container-runtime/src/test/garbageCollection.spec.ts

```diff
@@ -114,6 +115,134 @@ describe("Garbage Collection Tests", () => {
         sessionStorageConfigProvider.value.getRawConfig = oldRawConfig;
     });
 
+    describe("Configuration", () => {
+        const createGcWithPrivateMembers = (gcMetadata?: IGCMetadata, gcOptions?: IGCRuntimeOptions, snapshotCacheExpiryMs?: number) => {
+            const metadata: IContainerRuntimeMetadata | undefined = gcMetadata && { summaryFormatVersion: 1, message: undefined, ...gcMetadata };
+            const gcWithPrivates: IGarbageCollector & {
+                readonly gcEnabled: boolean;
+                readonly sweepEnabled: boolean;
+                readonly shouldRunGC: boolean;
+                readonly shouldRunSweep: boolean;
+                readonly trackGCState: boolean;
+                readonly testMode: boolean;
+                readonly latestSummaryGCVersion: GCVersion;
+                readonly sessionExpiryTimeoutMs: number | undefined;
+                readonly inactiveTimeoutMs: number;
+                readonly sweepTimeoutMs: number | undefined;
+            } = createGarbageCollector({ metadata, gcOptions, snapshotCacheExpiryMs }) as any;
+            return gcWithPrivates;
+        };
+        describe("Existing container", () => {
+            it("No metadata", () => {
+                const gc = createGcWithPrivateMembers({});
+                assert(!gc.gcEnabled, "gcEnabled incorrect");
+                assert(!gc.sweepEnabled, "sweepEnabled incorrect");
+                assert(gc.sessionExpiryTimeoutMs === undefined, "sessionExpiryTimeoutMs incorrect");
+                assert(gc.sweepTimeoutMs === undefined, "sweepTimeoutMs incorrect");
+                assert.equal(gc.latestSummaryGCVersion, 0, "latestSummaryGCVersion incorrect");
+            });
+            it("gcFeature 0", () => {
+                const gc = createGcWithPrivateMembers({ gcFeature: 0 });
+                assert(!gc.gcEnabled, "gcEnabled incorrect");
+                assert.equal(gc.latestSummaryGCVersion, 0, "latestSummaryGCVersion incorrect");
+            });
+            it("gcFeature 0, sweepEnabled true", () => {
+                const gc = createGcWithPrivateMembers({ gcFeature: 0, sweepEnabled: true });
+                assert(!gc.gcEnabled, "gcEnabled incorrect");
+                assert(gc.sweepEnabled, "sweepEnabled incorrect");
```

Comment:
Sweep should not be running if gc is not enabled.

Reply:
Ultimately, sweep won't run but the file will be marked as able to run sweep. It may actually be correct. Not entirely sure, will hopefully clear up in subsequent PRs in this space.

Reply:
Oh wait, I'm thinking about `shouldRunSweep`